servers/visual/shader_language.cpp

```diff
@@ -3279,8 +3279,10 @@ ShaderLanguage::Node *ShaderLanguage::_parse_expression(BlockNode *p_block, cons
 								return NULL;
 							}
 						} else {
-							_set_error("Only integer constants are allowed as index at the moment");
-							return NULL;
+							if (index->type != Node::TYPE_VARIABLE && index->type != Node::TYPE_ARRAY && index->type != Node::TYPE_OPERATOR) {
+								_set_error("Only integer expressions are allowed as index at the moment");
```

Comment:
I'd remove "at the moment". Or do we plan to add support for more types?

Reply:
Fixed.